src/go/rpk/pkg/config/config.go

```diff
@@ -48,7 +48,8 @@ func Default() *Config {
 			DeveloperMode: true,
 		},
 		Rpk: RpkConfig{
-			CoredumpDir: "/var/lib/redpanda/coredump",
+			CoredumpDir:     "/var/lib/redpanda/coredump",
+			Overprovisioned: true,
```

Comment:
Reason:

- Default() sets Redpanda.DeveloperMode: true.
- Redpanda.DeveloperMode sets --sandbox which sets Overprovisioned.
- `rpk redpanda mode developer` also sets Overprovisioned

Seems logic to include it upfront in the default config.